### PR TITLE
Update client

### PR DESCRIPTION
**Pull request by jmattheis**
You can ignore https://github.com/gotify/go-api-client/commit/4321f1f20dbedee2bbefbbd32b38f50e18d5652f

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,5 +1,5 @@
-GOTIFY_VERSION=1.2.0
-PACKAGE=github.com/gotify/go-api-client
+GOTIFY_VERSION=2.0.4
+PACKAGE=github.com/gotify/go-api-client/v2
```
Comment (eternal-flame-AD) [inline Makefile]: Is this variable used anywhere?

File: Makefile [hunk 2]
```diff
@@ -15,22 +15,22 @@ clean-tools:
 
 get-tools: clean-tools
 	mkdir .tools || true
-	wget -O ${SWAGGER} https://github.com/go-swagger/go-swagger/releases/download/v0.17.2/${DOWNLOAD_SWAGGER}
+	wget -O ${SWAGGER} https://github.com/go-swagger/go-swagger/releases/download/v0.19.0/${DOWNLOAD_SWAGGER}
```
Comment (eternal-flame-AD) [inline Makefile:17]: Why not use `go get -u` as we did so in the main repo?

File: Makefile [hunk 3]
```diff
@@ -15,22 +15,22 @@ clean-tools:
...
 	mkdir build || true
-	wget -O ${TEMP_SPEC} https://raw.githubusercontent.com/gotify/server/v${GOTIFY_VERSION}/docs/spec.json 
+	wget -O ${TEMP_SPEC} https://raw.githubusercontent.com/gotify/server/v${GOTIFY_VERSION}/docs/spec.json
 
 generate: obtain-spec
 	${SWAGGER} generate client -f ${TEMP_SPEC} --additional-initialism=rest --skip-models
-	find client/. -type f -exec sed -i 's/github.com\/gotify\/go-api-client\/models/github.com\/gotify\/server\/model/g' {} +
 
 test:
 	${SWAGGER} generate client --help
```
Comment (eternal-flame-AD) [inline Makefile]: This always yields a non-zero result, is it used anywhere?

File: go.mod [hunk 4]
```diff
@@ -1,4 +1,4 @@
-module github.com/gotify/go-api-client
+module github.com/gotify/go-api-client/v2
```
Comment (eternal-flame-AD) [inline go.mod:1]: `go mode tidy` yields a diff on my machine, please update go mod requirements.
```diff
diff --git a/go.mod b/go.mod
index 59a44f9..f7232c7 100644
--- a/go.mod
+++ b/go.mod
@@ -5,6 +5,4 @@ require (
        github.com/go-openapi/runtime v0.17.2
        github.com/go-openapi/strfmt v0.17.0
        github.com/go-openapi/swag v0.17.0
-       github.com/go-openapi/validate v0.17.0
-       golang.org/x/net v0.0.0-20181005035420-146acd28ed58
 )
```

**Review by eternal-flame-AD — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (jmattheis) [reply Makefile:17]: I wanted to specify an exact version, I guess when I did this there was a bug in the most recent version.

Comment (eternal-flame-AD) [reply Makefile:17]: Okay.

**Review by eternal-flame-AD — APPROVED** [review]

**Pull request merged**